Userland/Libraries/LibCore/System.cpp

```diff
@@ -245,6 +255,28 @@ ErrorOr<void> munmap(void* address, size_t size)
     return {};
 }
 
+ErrorOr<int> anon_create([[maybe_unused]] size_t size, [[maybe_unused]] int options)
+{
+    int fd = -1;
+#if defined(__serenity__)
+    fd = ::anon_create(size, options);
+#elif defined(__linux__)
+    // FIXME: Support more options on Linux.
+    auto linux_options = ((options & O_CLOEXEC) > 0) ? MFD_CLOEXEC : 0;
+    fd = memfd_create("", linux_options);
+    if (fd < 0)
+        return Error::from_errno(errno);
+    if (::ftruncate(fd, size) < 0) {
+        auto saved_errno = errno;
+        TRY(close(fd));
+        return Error::from_errno(saved_errno);
```

Comment:
Would be nice if we have a way to combine multiple error.
Kinda like `go.uber.org/multierr` in go.
```go
import "go.uber.org/multierr"

// ...
return multierr.Combine(
    close(fd),
    Error.from_errno(saved_errno),
)
```
Because for example, right now if `close` fail for example, we loose the `ftruncate` error.
(which you might argue it doesn't matter since in both cases an error is an error, I just find weird that if close and ftruncate fail we tell that instead of the root cause which is ftruncate)

Reply:
I don't think we need that right now. Please don't refer to outside code, refer to other idiomatic code when complaining :^). And the close call failing doesn't matter, because if we fail to close, we probably failed to open in the first place.

Reply:
I've should have been more clear that I don't think this was worth thinking. :)

I was just saying that for later if anyone want to know why this is happening that way.